### ISSUE TITLE
Easier filter configuration

### ISSUE THREAD

**Issue by JeffGillman** [zumoshi/BrowserSelect#9]
To help non-technical people use BrowserSelect: I'd suggest that when the pick-a-browser screen shows up, there is an "Always" button for each browser that will automatically add the filter for *.domain.tld to the filters.

**Comment by zumoshi** [comment]
Hmm, good idea. I'll see if I can also add this while doing the other changes you mentioned.

**Comment by zumoshi** [comment]
so far the settings menu got an Apply button, added some extra description in the Filter configurations and added BrowserSelect as an option to filters.

now about this suggestion, I added the checkbox but I think it kind of doesn't match the style of app.
![image](https://cloud.githubusercontent.com/assets/4184939/16665265/2d9b0bd0-4498-11e6-874c-ec887316207d.png)

would it be acceptable if instead of having a label for the checkbox I move the text to a tooltip ?
so there would be only  a small box in the right corner, similar to the image below:

![image](https://cloud.githubusercontent.com/assets/4184939/16665683/b859b176-4499-11e6-9d41-613722d01538.png)

(I'll probably remove the empty space in bottom though before releasing)

I'm also open to other suggestions , I'm not really an expert when it comes to UX design.

**Comment by zumoshi** [comment]
Reading through your suggestion again i realized you suggested a button for each browser not a checkbox. so something like this?

![image](https://cloud.githubusercontent.com/assets/4184939/16666071/46f120e4-449b-11e6-8254-772014028392.png)

or

![image](https://cloud.githubusercontent.com/assets/4184939/16666180/b136911e-449b-11e6-93d6-4289fdd85cee.png)


which of these three designs seem better so far? or is there another way to make it?

**Comment by JeffGillman** [comment]
The top one looks great to me. Exactly what I had envisioned. Cool!

What’s the “(1,f)” stuff?

**Comment by zumoshi** [comment]
those were there since the first version , they are the keys you can press to select each browser.
to select each browser other than clicking on it (obviously) you can press its index (counting from the left) or the first letter of each word in its name. 
so for firefox (its first browser) you can press `1` or `f` (first letter of Firefox) on the keyboard.

just to clarify , by "top one" you are referring to the first image of the second post or first post? the one with big buttons under each image?

**Comment by JeffGillman** [comment]
Correct, this one:

https://cloud.githubusercontent.com/assets/4184939/16666071/46f120e4-449b-11e6-8254-772014028392.png

With the big buttons under each image.

**Comment by zumoshi** [comment]
I've released a new version, please download and check to see if it meets all of your requirements.

**Comment by JeffGillman** [comment]
It looks great! I really think the changes are good. Always buttons work well.

For some reason it didn’t recognize Internet Explorer as an installed browser.





(Windows 7 64-bit, latest IE installed).

 

More feedback soon.

**Comment by zumoshi** [comment]
that is weird, I had another issue before that said it didn't detect a portable browser, but it shouldn't have any problems detecting IE, I tested it on multiple systems with windows 7, 8, 8.1 and 10 and it always detected IE.
did the last version detect it ?

also , could you please check contents of the following key in the registry ?
```
HKEY_LOCAL_MACHINE\SOFTWARE\Clients\StartMenuInternet```

**Comment by JeffGillman** [comment]
I had never tried BrowserSelect on that computer before. I uninstalled the old version on *this* computer, installed the new one, and it detects IE (and Chrome and Firefox) correctly, as it did before. This machine is also Win 7 64-bit with current IE.  I will check that registry key when I get back to my other computer. Note: I sent an email concerning another issue to your email address as specified in BrowserSelect's About screen, just in case you don't check that very often.

**Comment by zumoshi** [comment]
yes, that's how it should look like, just to clarify my app doesn't create these keys, they are made by Windows and are used to display the list of available browsers when choosing the default one.

if IE is missing from there then the problem is not with my app but rather with that os installation.
even though I've never experienced this personally but some after googling it seems upgrading IE sometimes may have this side effect. you can read [this article](https://www.question-defense.com/2009/07/25/windows-internet-explorer-icon-missing-from-start-menu) or similar ones.

basically you have to add a key for each browser that fails to get detected there and it will detect it.

**Comment by JeffGillman** [comment]
The computer that doesn't show IE in BrowserSelect (I believe the issue is the very last line, or rather the lack of it, as compared to the above):

```
Windows Registry Editor Version 5.00

[HKEY_LOCAL_MACHINE\SOFTWARE\Clients\StartMenuInternet\IEXPLORE.EXE]
@="Internet Explorer"
"LocalizedString"="@C:\\Program Files\\Internet Explorer\\iexplore.exe,-702"

[HKEY_LOCAL_MACHINE\SOFTWARE\Clients\StartMenuInternet\IEXPLORE.EXE\DefaultIcon]
@="C:\\Program Files\\Internet Explorer\\iexplore.exe,-9"

[HKEY_LOCAL_MACHINE\SOFTWARE\Clients\StartMenuInternet\IEXPLORE.EXE\InstallInfo]
"IconsVisible"=dword:00000001
"ShowIconsCommand"="\"C:\\Windows\\System32\\ie4uinit.exe\" -show"
"ReinstallCommand"="\"C:\\Windows\\System32\\ie4uinit.exe\" -reinstall"
"HideIconsCommand"="\"C:\\Windows\\System32\\ie4uinit.exe\" -hide"

[HKEY_LOCAL_MACHINE\SOFTWARE\Clients\StartMenuInternet\IEXPLORE.EXE\shell]

[HKEY_LOCAL_MACHINE\SOFTWARE\Clients\StartMenuInternet\IEXPLORE.EXE\shell\naom]
"MUIVerb"="@C:\\Windows\\System32\\ieframe.dll,-39229"

[HKEY_LOCAL_MACHINE\SOFTWARE\Clients\StartMenuInternet\IEXPLORE.EXE\shell\naom\command]
@="\"C:\\Program Files\\Internet Explorer\\iexplore.exe\" -extoff"

[HKEY_LOCAL_MACHINE\SOFTWARE\Clients\StartMenuInternet\IEXPLORE.EXE\shell\open]

[HKEY_LOCAL_MACHINE\SOFTWARE\Clients\StartMenuInternet\IEXPLORE.EXE\shell\open\command]
```

So this appears to be some configuration issue with my computer, and not related to BrowserSelect.

**Comment by zumoshi** [comment]
yes , that key is where browser select gets the executable path for browser. if its not there it can't run the browser, so it won't display it.
since those keys are not made by BrowserSelect, yes that is a configuration issue with that particular windows instalation. it is easy to fix though, just add the path to IE's exe file to that key.

**Comment by JeffGillman** [comment]
I just noticed that even though I'm using the Always buttons for sites, I'm being asked again to choose a Browser each time I open most sites. So I checked settings, and see that in many cases, the filter being created is missing the period before the TLD. For example, I have multiple identical filters for *.yammercom, instead of *.yammer.com. But the one I just created via Always for GitHub is correct at *.github.com. And I think I figured it out: if the URL contains a subdomain, the filter is created incorrectly. If I go to http://justatest.github.com, the filter is created as *.githubcom .
